File: fingerprint/src/main/java/com/yf/verify/fingerprint/factory/BaseBiometricPrompt28.java
package com.yf.verify.fingerprint.factory;

import android.support.v4.app.FragmentActivity;
import android.support.v4.hardware.fingerprint.FingerprintManagerCompat;

import com.yf.verify.callback.FingerprintAuthenticatedCallback;
import com.yf.verify.fingerprint.FingerprintBottomDialogFragment;

import javax.crypto.Cipher;

public class BaseBiometricPrompt28 implements BaseBiometricPrompt {
    private FragmentActivity activity;
    private Cipher cipher;
    private FingerprintManagerCompat mFingerprintManager;
    private FingerprintAuthenticatedCallback fingerprintCallback;

    public BaseBiometricPrompt28(FragmentActivity activity, Cipher cipher, FingerprintManagerCompat mFingerprintManager, FingerprintAuthenticatedCallback fingerprintCallback) {
        this.activity = activity;
        this.cipher = cipher;
        this.mFingerprintManager = mFingerprintManager;
        this.fingerprintCallback = fingerprintCallback;
    }

    @Override
    public void show() {
        if (null == mFingerprintManager) {
            if (null != fingerprintCallback) {
                fingerprintCallback.onNoEnrolledFingerprints();
            }
            return;
        }
        FingerprintBottomDialogFragment fragment = FingerprintBottomDialogFragment.newInstance();
        fragment.setFingerprintManager(mFingerprintManager);
        fragment.setCryptoObject(new FingerprintManagerCompat.CryptoObject(cipher));
        fragment.setCallback(fingerprintCallback);
        fragment.show(activity.getSupportFragmentManager(), "FingerprintBottomDialogFragment");
    }
}

File: fingerprint/src/main/java/com/yf/verify/fingerprint/factory/BaseBiometricPrompt28.kt
package com.yf.verify.fingerprint.factory

import android.annotation.TargetApi
import android.hardware.fingerprint.FingerprintManager
import android.os.Build
import android.support.v4.app.FragmentActivity
import com.yf.verify.callback.FingerprintAuthenticatedCallback
import com.yf.verify.callback.UpdateFingerprintCallback
import com.yf.verify.fingerprint.FingerprintAuthenticationDialogFragment
import javax.crypto.Cipher

@TargetApi(Build.VERSION_CODES.M)
class BaseBiometricPrompt28(var activity: FragmentActivity, var cipher: Cipher, var stage: FingerprintAuthenticationDialogFragment.Stage,
                            var updateFingerprintCallback: UpdateFingerprintCallback, var fingerprintCallback: FingerprintAuthenticatedCallback) : BaseBiometricPrompt {

    override fun show() {
        val fragment = FingerprintAuthenticationDialogFragment.newInstance()
        fragment.setCryptoObject(FingerprintManager.CryptoObject(cipher))
        fragment.setStage(stage)
        fragment.setUpdateFingerprintCallback(updateFingerprintCallback)
        fragment.setCallback(fingerprintCallback)
        fragment.show(activity?.supportFragmentManager!!, "FingerprintAuthenticationDialogFragment")
    }

}